### PR TITLE
Added `error = FALSE` argument to `if_else2()`

### PR DESCRIPTION
**Pull request by dunkenwg**
Also added more `if_else2()` tests

### CODE REVIEW

**Comment by dunkenwg** [comment]
Failing on macos because it can't install R 4.4.2

File: R/if-else2.R [hunk 1]
```diff
@@ -34,8 +35,8 @@
 #' dplyr::mutate(data,
 #'   x3 = dplyr::if_else(str_detect2(y, "x is false"), FALSE, x)
 #' )
-if_else2 <- function(condition, true, false) {
-  if (!any(condition)) {
+if_else2 <- function(condition, true, false, error = FALSE) {
```
Comment (joethorley) [inline R/if-else2.R:38]:
```suggestion
if_else2 <- function(condition, true, false, error = FALSE) {
	chk_flag(error)
```

**Review by joethorley — CHANGES_REQUESTED** [review]
I think it is worth type checking the error argument (see suggested change.)

Also the error argument needs adding to the documentation.

**Comment by dunkenwg** [comment]
The error argument should be in the documentation

**Review by joethorley — APPROVED** [review]

**Comment by dunkenwg** [comment]
@aylapear, would you be able to take a look at this and I'll merge it in?

**Review by aylapear — APPROVED** [review]

**Pull request merged**